src/components/NewsletterSignup/NewsletterSignup.tsx

```diff
@@ -0,0 +1,195 @@
+   * field submitted data. */
+  hiddenFields?: any;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Toggles the invalid state for the email field. */
+  isInvalidEmail?: boolean;
+  /* Optional value to determine the section color highlight */
+  newsletterSignupType?: SectionTypes;
+  /** A submit handler function that will be called when the form is submitted. */
+  onSubmit: (event: React.FormEvent<any>) => void;
+  /** A on change handler function for the text input. */
+  onChangeEmail: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** The value of the email text input field. */
+  valueEmail: string;
+  /** Used to populate the `<h3>` header title. */
+  title: string;
+  /** Used to specify what is displayed in the component content area. */
+  view?: "form" | "submitting" | "confirmation" | "error";
+}
+
+/**
+ * The NewsletterSignup component provides a way for patrons to register for an
+ * email-based newsletter distribution list.
+ *
+ * The component can show four different views, depending on the state of the
+ * email submission: form, loading, confirmation, and error.
+ */
+export const NewsletterSignup = chakra(
+  forwardRef<any, NewsletterSignupProps>(
+    (
+      {
+        className,
+        confirmationText = "Thank you! You have successfully subscribed to our email updates! You can update your email subscription preferences at any time using the links at the bottom of the email.",
+        descriptionText = "Stay connected with the latest research news from NYPL, including information about our events, programs, exhibitions, and collections.",
+        formHelperText,
+        hiddenFields,
+        id,
+        isInvalidEmail = false,
+        newsletterSignupType = "whatsOn",
+        onChangeEmail,
+        valueEmail,
+        onSubmit,
+        title = "Sign Up for Our Newsletter!",
+        view = "form",
+        ...rest
+      },
+      ref?
+    ) => {
+      const { isLargerThanMobile } = useNYPLBreakpoints();
+      const styles = useStyleConfig("NewsletterSignup", {});
+      const iconColor = useColorModeValue(null, "dark.ui.typography.body");
+
+      const isFormView = view === "form" || view === "submitting";
+
+      // Manage focus to ensure accessibility when confirmation or error message is rendered.
+      const focusRef = React.useRef<HTMLDivElement>(null);
+      // When view prop changes, set focus to the confirmation or error content element.
+      React.useEffect(() => {
+        focusRef.current?.focus();
+      }, [view]);
+
+      return (
+        <Stack
+          direction={isLargerThanMobile ? "row" : "column"}
+          ref={ref}
+          __css={styles}
+          {...rest}
+        >
+          <Stack direction={isLargerThanMobile ? "row" : "column"} id={"info"}>
+            <Box
+              bg={getSectionColors(newsletterSignupType, "primary")}
+              id={"color-box"}
```

Comment:
Not sure what the convention is here - should this be camel case?

Reply:
Noticed kebab-case in other places in the DS so ignore this!

Reply:
CSS class and id naming convention [seems to prefer hyphens](https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/).